sdcm/cluster.py

```diff
@@ -2752,6 +2755,17 @@ def is_cqlsh_support_cloud_bundle(self):
             )
         )
 
+    @property
+    def is_replacement_by_host_id_supported(self):
+        # NOTE: compare 'LegacyVersion' returns 'False' for the following comparison:
+        #         LegacyVersion("5.2.0") > LegacyVersion('5.2.0~dev')
+        #       and 'True' for the following:
+        #         LegacyVersion("5.2.0~rc1") > LegacyVersion('5.2.0~dev')
+        #       So, compare to '5.1.999' because any 5.2+ version provides 'True' in this case
+        if self.is_enterprise:
+            return LegacyVersion(self.scylla_version) > LegacyVersion('2022.2.999')
```

Comment:
isn't it better to be more explicit, and test with greater or equal?

```suggestion
            return LegacyVersion(self.scylla_version) >= LegacyVersion('2023.1.0')
```

Reply:
The answer is written in the comment above the code.
Using your variant any `dev` and `rc` versions will return `False` result. The current one will return correct `True` result in this case.

Reply:
I think we should be using `parse_scylla_version`

Reply:
Can we check for it in help or scylla.yaml instead?
Every time someone changes the versions format or something like this we discover too late that these things stopped working.

Reply:
I plan upload the version comparison solution today.
It will allow us to stop using `LegacyVersion` which blocks us from usage of latest `packaging` package.
And will correctly compare versions.

Reply:
roy's suggestion is also quite valid in this case, and we already have the code the does handle the `scylla --help` output, and should be more robust 

and we still need to handle versions correctly for other cases where the difference is not manifested in `scylla --help`

Reply:
We will talk about it in a PR that will be uploaded.
For the moment I can say that I added support to the most, known to me, "parsing" specifics the way that
"scylla-binary" and "docker tag" will be easily comparable using new feature that will be uploaded.

Reply:
It is here: https://github.com/scylladb/scylla-cluster-tests/pull/5817